boot/freeldr/freeldr/arch/realmode/relocldr.inc

```diff
@@ -0,0 +1,68 @@
+/*
+ * PROJECT:     FreeLoader
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Add some helpers functions and flags necessary for the relocation.
+ * COPYRIGHT:   Copyright 2024 Daniel Victor <ilauncherdeveloper@gmail.com>
+ */
+
+FirstBoot:
+    .byte 1
+
+CheckNtldrBoot:
+    /* Don't clear the necessary variables if it's not the first boot */
+    cmp byte ptr ds:[FirstBoot], 0
+    jz .IsNotTheFirstBoot
+
+    /* Clear the necessary variables */
+    xor eax, eax
+    mov byte ptr ds:[BSS_BootedViaNtldr], al
+    mov dword ptr ds:[BSS_NtldrAddress], eax
+    mov dword ptr ds:[BSS_NtldrAddress + 4], eax
+    mov byte ptr ds:[FirstBoot], al
+
+.IsNotTheFirstBoot:
```

Comment:
`.isNotTheFirstBoot:` with first letter in lowercase, as you seem to use for internal labels? (if so, adjustment in the jz above would be required)